jOOQ/src/main/java/org/jooq/impl/JoinTable.java

```diff
@@ -562,7 +558,14 @@ else if (rightToLeft.size() == 1 && leftToRight.size() == 0) {
             return onKey((ForeignKey<?, ?>) rightToLeft.get(0), rhs, lhs);
         }
 
-        throw onKeyException();
+        if (rightToLeft.isEmpty() && leftToRight.isEmpty()) {
+            throw new DataAccessException("No matching Key found between tables " + lhs + " and " + rhs);
+        } else {
+            throw new DataAccessException(
+                    "`onKey()` was ambiguous between tables " + lhs + " and " + rhs + ".\n" +
+                            "Found possible keys: " + leftToRight + " and " + rightToLeft + ".\n" +
+                            "Please use .onKey(ForeignKey) instead.");
+        }
```

Comment:
I see what you mean. I'll adapt this code in two ways, though:

1. I'll factor out the different messags back into that utility method, to increase reuse, because:
2. I suspect that ambiguity can also happen with the other ways of using `onKey()` in more complicated setups